spyder/plugins/updatemanager/workers.py

```diff
@@ -250,16 +277,9 @@ def _download_installer(self):
     def _clean_installer_path(self):
         """Remove downloaded file"""
         if osp.exists(self.installer_path):
-            try:
-                shutil.rmtree(self.installer_path)
-            except OSError as err:
-                logger.debug(err, stack_info=True)
-
+            os.remove(self.installer_path)
         if osp.exists(self.installer_size_path):
-            try:
-                shutil.rmtree(self.installer_size_path)
-            except OSError as err:
-                logger.debug(err, stack_info=True)
```

Comment:
I'd prefer to leave the previous approach here because it captures an inconsequential error message but logs it for debugging purposes.

Reply:
I reverted @dalthviz's change because `self.installer_path` is a file, not a directory, so `shutil.rmtree` fails. I thought this was an overisight. I suggest the following instead.
```python
        installer_dir = osp.dirname(self.installer_path)
        if osp.exists(installer_dir):
            try:
                shutil.rmtree(installer_dir)
            except OSError as err:
                logger.debug(err, stack_info=True)
```